src/recurrenceIterator.test.ts

```diff
@@ -292,22 +306,25 @@ describe('recurrenceIterator', () => {
     const seq3 = {
       start_datetime: '2017-01-01T23:00Z',
       end_datetime: '2017-01-02T23:00Z',
-      recurrence_rule: { freq: 'weekly', byday: ['TH', 'SU'] },
+      recurrence_rule: {
+        freq: Frequency.WEEKLY,
+        byday: [WeekDay.TH, WeekDay.SU],
+      },
     };
-    let ri = recurrenceIterator([seq1], '2017-01-01T00:00Z');
+    let ri = recurrenceIterator([seq1], new Date('2017-01-01T00:00Z'));
```

Comment:
The usage here was ambiguous and a little concerning, since inputted strings are compared directly against `Date` under the hood. I opted to constrain inputs to `Date` instances, which adds safety without needing to change implementation. Updated tests to conform to the stricter contract.